Libs/GObject/Structs/Type.cs

```diff
@@ -49,6 +49,25 @@ public Type(ulong value)
 
         #endregion
 
+        // Print out the name of the gtype (for debugging purposes)
+        public override string ToString()
```

Comment:
Can we use a separate function for this? Perhaps it is even only compiled into the binary with the `#define DEBUG` compile time constant?

To have a native C# behaviour I think we should keep our class names and the default implementation of the `ToString` method clean.

Reply:
I'm not sure I fully agree with this.

`ToString` is generally used to give a more human readable representation of an object, so it makes sense to me to print the GType class name instead of the type id number, which doesn't tell us all that much (e.g. when passing it to Console.WriteLine, etc).

That comment is a mistake from when I first wrote it, as it's not really being used for debugging but rather as a more general logging/introspection tool.

I'm not opposed to making this a separate function, but then we lose the ability to simply plug the type into `Console.WriteLine` and/or read it in the debugger.

Thoughts?

Reply:
You are right. Actually I did not test the output of this. But the default implementation should just return e.g. "Button" or not? I assumed that with your change it returns s.th. like "GButton" which is not matching the class name anymore. So I can be wrong here.

Will check later or perhaps you have some more information?

Reply:
Since it's the GType struct rather than the object, I imagine it would simply return the string "Type" whenever we call `ToString()` which doesn't seem all that useful (I haven't actually checked).

With this version it returns the name of the GType straight from GLib (so if we introspected a type id of e.g. 107, we'd be returned a string that says "GtkButton"), which makes the most sense for logging.

Reply:
Oh I was completely in the wrong context 🙈 . This is the type class, my mind was in the object class.

Reply:
I just used http://try.dot.net with `Console.WriteLine(typeof(System.IO.File).ToString());` and it returns  `
System.IO.File` So I agree to your change completely. Thanks for claryfing. I'm going to merge :+1: